conans/util/files.py

```diff
@@ -44,6 +44,8 @@ def remove_if_dirty(item):
             else:
                 rmdir(item)
         clean_dirty(item)
+        return True
+    return False
```

Comment:
Let's the calling method know if a removal happened, useful when skipping dirty files in the backup upload file gathering